src/dotnet-install.sh

```diff
@@ -753,9 +753,8 @@ download() {
         elif machine_has "wget"; then
             downloadwget "$remote_path" "$out_path" || failed=true
         else
-            unset http_code
-            download_error_msg="Missing dependency: neither curl nor wget was found."
-            break
+            say_err "Missing dependency: neither curl nor wget was found."
```

Comment:
I'd recommend that you return to the old logic here. There should be no `http_code` if there's no http request.

Reply:
Do you mean that I should be unsetting it? We are exiting the script when we hit this condition though, so that doesn't matter unless I am missing something? `http_code` is only a global and not an env var.

FWIW it appears that the script was already exiting if neither curl or wget were present thanks to `check_min_reqs()`.

Reply:
Ah, then maybe this is defensive. return 1 wouldn't exit here, but if check_min_reqs does it, then your original fix was better. Just means that the no-wget-or-curl branch of this was dead all along. Thanks for explaining.

Reply:
Yep, I missed that about returning 1 from inside the function not actually exiting, thank you for catching that. I think it makes sense to be defensive also even if the branch is dead, and I think it's better to just exit with the message logged so I updated the PR to that fwiw.